### PR TITLE
[Shipping] Refactor shipment state machine

### PR DESCRIPTION
**Pull request by Arminek**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | yes
| Related tickets | 
| License         | MIT

### CODE REVIEW

File: features/order/managing_orders/placing_an_order_with_shipment.feature [hunk 1]
```diff
@@ -0,0 +1,21 @@
+@managing_orders
+Feature: Payments are in the state "new" after checkout
+    In order to correctly process customer's payments
+    As an Administrator
+    I want to have new payments after my customer's checkout
```
Comment (michalmarcinkowski) [inline features/order/managing_orders/placing_an_order_with_shipment.feature]: The whole description is incorrect 😕

File: features/order/managing_orders/placing_an_order_with_shipment.feature [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        And the store ships everywhere for free
+        And the store allows paying with "Cash on Delivery"
+        And there is a customer "lucy@teamlucifer.com" that placed an order "#00000666"
+        And the customer bought a single "Angel T-Shirt"
+        And the customer "Lucifer Morningstar" addressed it to "Seaside Fwy", "90802" "Los Angeles" in the "United States" with identical billing address
+        And the customer chose "Free" shipping method with "Cash on Delivery" payment
+        And I am logged in as an administrator
+
+    @ui
+    Scenario: Checking payment state of a placed order
```
Comment (michalmarcinkowski) [inline features/order/managing_orders/placing_an_order_with_shipment.feature]: Really? 😄

File: features/order/managing_orders/placing_an_order_with_shipment.feature [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        When I view the summary of the order "#00000666"
+        Then it should have shipment state "Ready"
```
Comment (michalmarcinkowski) [inline features/order/managing_orders/placing_an_order_with_shipment.feature]: Then it should have shipment **in** "Ready" state

File: src/Sylius/Component/Core/OrderProcessing/StateResolver.php [hunk 4]
```diff
@@ -86,11 +86,8 @@ protected function getShippingState(OrderInterface $order)
         $states = array_unique($states);
 
         $acceptableStates = [
```
Comment (michalmarcinkowski) [inline src/Sylius/Component/Core/OrderProcessing/StateResolver.php]: All code related to shipments should be removed from here.

Btw. We should remove this class, is it used anywhere?

Comment (Arminek) [inline src/Sylius/Component/Core/OrderProcessing/StateResolver.php]: It will be in another pr related with main shipping state of an order.

File: features/order/managing_orders/placing_an_order_with_shipment.feature [hunk 5]
```diff
@@ -0,0 +1,21 @@
+@managing_orders
+Feature: Shipments are in the state "ready" after checkout
```
Comment (michalmarcinkowski) [inline features/order/managing_orders/placing_an_order_with_shipment.feature]: Feature description should be less technical and as near to business language as possible e.g. `Shipments should be ready to send after checkout`

**Comment by michalmarcinkowski** [comment]
Thank you Arek!

File: src/Sylius/Behat/Context/Ui/Admin/ManagingOrdersContext.php [hunk 6]
```diff
@@ -571,4 +571,15 @@ public function iShouldSeeOrderWithNumber($orderNumber)
             sprintf('Cannot find order with "%s" number in the list.', $orderNumber)
         );
     }
+
+    /**
+     * @Given it should have shipment in state :shipmentState
```
Comment (lchrusciel) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingOrdersContext.php]: `Given` -> `Then`
